views/Graph.react.js

```diff
@@ -214,48 +240,67 @@ class Graph extends Component {
   handleResize = () => {
     this.setState({
       graphWidth: window.innerWidth,
-      graphHeight: window.innerHeight - 30
+      graphHeight: window.innerHeight - 30,
     })
   }
   resync = () => {
-    window.fetch(this.state.socketIOSubjectRoom, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      credentials: 'same-origin'
-    }).then((response) => {
-      if (response.status !== 201) {
-        return
-      }
-      return response.json()
-    }).then((response) => {
-      this.setState({ taskId: response.correlationId, loading: true, success: false })
-    })
+    window
+      .fetch(this.state.socketIOSubjectRoom, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+      })
+      .then((response) => {
+        if (response.status !== 201) {
+          return
+        }
+        return response.json()
+      })
+      .then((response) => {
+        this.setState({
+          taskId: response.correlationId,
+          loading: true,
+          success: false,
+        })
+      })
   }
   closeStat = () => {
     this.setState({
       openStat: false,
-    });
+    })
   }
   openStat = () => {
     this.setState({
       openStat: true,
-    });
-  }
-  componentDidUpdate() {
+    })
   }
+  componentDidUpdate() {}
   // eslint-disable-next-line react/no-deprecated
   async componentWillMount() {
     try {
-      const acl = await fetchSubjects();
-      this.setState(getCounts({ acl }));
+      const acl = await fetchSubjects()
+      this.setState(getCounts({ acl }))
+      const preferences = await fetchPreferences(this.props.user.uid)
+      this.setState({ preferences })
+      const configurations = await fetchConfigurations(this.props.user.uid)
+      this.setState(() => ({ configurationsList: configurations }))
```

Comment:
we should group these setState calls together if we can

```suggestion
      const [acl, preferences, configurations] = await Promise.all([
        fetchSubjects(),
        fetchPreferences(this.props.user.uid),
        fetchConfigurations(this.props.user.uid),
      ])

      this.setState({
        ...getCounts({ acl }),
        preferences,
        configurationsList: configurations,
      })
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/129/commits/75fa4e051ac01363f4567f566d2f67ddf2280f34